bpf/bpf_host.c

```diff
@@ -1651,7 +1651,23 @@ int cil_to_host(struct __ctx_buff *ctx)
 	 * to mark as PACKET_OTHERHOST and drop.
 	 */
 	ctx_change_type(ctx, PACKET_HOST);
-#endif
+
+# ifdef ENABLE_NODEPORT
+	if (!validate_ethertype(ctx, &proto)) {
+		ret = DROP_UNSUPPORTED_L2;
+		goto out;
```

Comment:
Trying to understand the scope / what traffic this captures. Can we limit the impact of this at all - and *only* handle the specific traffic type that we know is broken at the moment? Ideally everything else would still float into `to-netdev`, and get RevDNAT there as before.

Reply:
Make sense, I can wrap it with `if (skb->mark & 0xf00 == 0xe00)` so only to-be-IPsec-encrypted skbs are checking revDNAT.